syntax = "proto3";

package orderqueue;

service OrderQueueService {
    rpc Enqueue (EnqueueRequest) returns (EnqueueResponse);
    rpc Dequeue (DequeueRequest) returns (DequeueResponse);
}

message Order {
    int32 orderId = 1;
    string userName = 2;
}

message EnqueueRequest {
    Order order = 1;
}

message EnqueueResponse {
    bool success = 1;
    string message = 2;
}

message DequeueRequest {
    int32 executor_id = 1;
}

message DequeueResponse {
    bool success = 1;
    Order order = 2;
}